### PR TITLE
apacheHttpd, nixos/httpd: set lua paths

### PR DESCRIPTION
**Pull request by nagy**
###### Motivation for this change
Without this, lua script handlers do not find the packages that are provided with the lua package.
Note that, since the `luaSupport` argument of the `apacheHttpd` derivation defaults to `false`, this wont change the generated config file unless the derivation is built with lua support enabled.

More information on the used apache directives is here:
  https://httpd.apache.org/docs/trunk/mod/mod_lua.html#luapackagepath
  https://httpd.apache.org/docs/trunk/mod/mod_lua.html#luapackagecpath
  

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nagy** [comment]
@lovek323 @peti , since you are the maintainers, could you review this please ?

File: nixos/modules/services/web-servers/apache-httpd/default.nix [hunk 1]
```diff
@@ -326,6 +333,8 @@ let
 
     ${sslConf}
 
+    ${if cfg.package.luaSupport then luaSetPaths else ""}
```
Comment (aanderse) [inline nixos/modules/services/web-servers/apache-httpd/default.nix:336]: @nagy this ends up being slightly annoying for me... do you mind rewriting this as
```
${optionalString (cfg.package.luaSupport or false) luaSetPaths}
```

**Review by aanderse — COMMENTED** [review]
After the fact, but I would like this amended if you don't mind please.

File: nixos/modules/services/web-servers/apache-httpd/default.nix (hunk 1 above)
Comment (nagy) [reply nixos/modules/services/web-servers/apache-httpd/default.nix:336]: I have adapted this in another pull request that I opened recently on this topic again.

https://github.com/NixOS/nixpkgs/pull/108511

Comment (aanderse) [reply nixos/modules/services/web-servers/apache-httpd/default.nix:336]: Thank you very much @nagy!

**Pull request merged**